beacon-chain/state/stategen/setter.go

```diff
@@ -58,6 +59,20 @@ func (s *State) saveStateByRoot(ctx context.Context, blockRoot [32]byte, state *
 	ctx, span := trace.StartSpan(ctx, "stateGen.saveStateByRoot")
 	defer span.End()
 
+	s.saveHotStateDB.lock.Lock()
+	if s.saveHotStateDB.enabled && state.Slot()%s.saveHotStateDB.duration == 0 {
```

Comment:
Check divide by zero. If `s.saveHotStateDB.duration` is zero then you're going to panic.
```suggestion
	if s.saveHotStateDB.enabled && state.Slot()%math.Max(s.saveHotStateDB.duration, 1) == 0 {
```